### DISCUSSION TITLE
Pick one relation

### DISCUSSION THREAD

**Discussion by segovo** [pocketbase/pocketbase#1798] [Q&A] [answered]
Hello!

I want to create a links collection, where you can create a link to either a Post, a User, or a Folder. How can I make it so that a link can only be created with one of those relation fields? So if the folder relation field has been filled in, it won't be possible to fill in the  Post or User relation fields?

My only current idea, is to create a filter rule like this: (post != "" && user = "" && folder = "") || (post = "" && user != "" && folder = "") || (post = "" && user = "" && folder != "").

The problem with this is that it means making a larger and larger and larger filter rule that will grow out of hand once I have 5 or 6 different things a link could point to. So, is there any other way to accomplish this?

**Comment by ganigeorgiev** [comment, marked as answer]
I don't think there is other easier way. Even on db level you can't define the constraint fields once because you still will have to check every other field, aka. there is no "only 1 field must be filled" constraint out of the box.

Alternatively, the check could be done programmatically in a before model (_or in a request hook if you want to allow admins to overwrite the behavior_)  when using [PocketBase as framework](https://pocketbase.io/docs/use-as-framework/), something like:

```go
package main

import (
	"errors"
	"log"

	"github.com/pocketbase/pocketbase"
	"github.com/pocketbase/pocketbase/core"
	"github.com/pocketbase/pocketbase/models"
)

func main() {
	app := pocketbase.New()

	app.OnModelBeforeCreate("links").Add(func(e *core.ModelEvent) error {
		record := e.Model.(*models.Record)

		vals := make([]string, 3)
		vals[0] = record.GetString("post")
		vals[1] = record.GetString("user")
		vals[2] = record.GetString("folder")

		nonEmpty := 0
		for _, v := range vals {
			if v != "" {
				nonEmpty++
			}
		}
		if nonEmpty != 1 {
			return errors.New("only one of post, user, folder fields could be set")
		}


		return nil
	})

	if err := app.Start(); err != nil {
		log.Fatal(err)
	}
}

```

**Reply by segovo** [reply]
The onModelBeforeCreate idea should work wonderfully for my situation, I really appreciate it, thank you!
